akka-typed-tests/src/test/java/akka/typed/javadsl/BehaviorBuilderTest.java

```diff
@@ -83,17 +83,22 @@ public void testImmutableCounter() {
 
     @Test
     public void testMutableCounter() {
-      Behavior<CounterMessage> mutable = Actor.mutable2(b -> {
+      Behavior<CounterMessage> mutable = Actor.mutable(ctx -> new Actor.MutableBehavior<CounterMessage>() {
         MutableStateHolder state = new MutableStateHolder();
 
-        return b.onMessage(Increase.class, (ctx, o) -> {
-          state.currentValue++;
-          return same();
-        })
-        .onMessage(Get.class, (ctx, o) -> {
-          o.sender.tell(new Got(state.currentValue));
-          return same();
-        });
+        @Override
+        public Receive<CounterMessage> createReceive() {
+          return receiveBuilder()
+            .onMessage(Increase.class, o -> {
```

Comment:
the idomatic way to write this is to delegate to a method, in my opinion, then no need for the state holder

```java
private Behavior<CounterMessage> receiveIncrease(Increase inc) {
  currentValue++;
  return this;
}
```

```
.onMessage(Increase.class, this::receiveIncrease)